FluentScheduler/Scheduler/InternalSchedule.cs

```diff
@@ -78,16 +78,20 @@ internal void Stop(bool block, int? timeout = null)
                 return;
 
             _tokenSource.Cancel();
-            _tokenSource.Dispose();
-
-            if (block && timeout.HasValue)
-                _task.Wait(timeout.Value);
-
-            if (block && !timeout.HasValue)
-                _task.Wait();
+            try
+            {
+                if (block && timeout.HasValue)
+                    _task.Wait(timeout.Value);
 
-            _task = null;
-            _tokenSource = null;
+                if (block && !timeout.HasValue)
+                    _task.Wait();
+            }
+            finally
+            {
+                _tokenSource.Dispose();
+                _task = null;
+                _tokenSource = null;
+            }
```

Comment:
Good catch. I might apply this change in a separate commit and test it alone before merging this pr.